src/input/datetime.rs

```diff
@@ -496,6 +496,11 @@ impl TzInfo {
         None
     }
 
+    fn fromutc<'p>(&self, py: Python<'p>, dt: &'p PyDateTime) -> PyResult<&'p PyDateTime> {
```

Comment:
```suggestion
    fn fromutc<'py>(&self, py: Python<'py>, dt: &'py PyDateTime) -> PyResult<&'py PyDateTime> {
```
use use `py` everywhere.